sklearn/tests/test_random_projection.py

```diff
@@ -222,13 +225,15 @@ def test_random_projection_transformer_invalid_input():
 
 
 def test_try_to_transform_before_fit():
+    data, _ = make_sparse_random_data(sp.coo_array, n_samples, n_features, n_nonzeros)
```

Comment:
This is good enough to check that we raise an error before `fit`. Also it makes me think that the random projections are not raising a `NotFittedError` apparently. This is weird.


```suggestion
    data, _ = make_sparse_random_data(sp.coo_matrix, n_samples, n_features, n_nonzeros)
```

Reply:
I understand. And we're using `sp.coo_matrix` here, because the CI will also use older scipy versions for testing?

They do use `check_is_fitted` and raise an `NotFittedError`, I have substituted the error types.